bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/SlicedProgressMonitor.java

```diff
@@ -86,6 +86,7 @@ public boolean isCanceled() {
 
 	@Override
 	public void setCanceled(boolean value) {
+		monitor.setCanceled(value);
```

Comment:
if you want this you need to `syncronize` on the monitor instance, beside that one usually don't want to cancel a slice, what is the use-case for that? According to the javadoc a monitor passed to a method should actually never call cancel so I'm not sure what these cancel tests "test"?

Maybe you want a split instead?